src/app/core/service-provider/products.service-provider.spec.ts

```diff
@@ -0,0 +1,149 @@
+
+      expect(result).toBe(instance(sparqueProductsServiceMock));
+    });
+
+    it('should return SparqueProductsService when Sparque config exists and sparque_search feature is enabled', () => {
+      configureTestBed({ apiKey: 'test-key' });
+      when(featureToggleServiceMock.getFeatures$('sparque_' as FeatureToggleType)).thenReturn(
+        of(['sparque_search'] as FeatureToggleType[])
+      );
+
+      const result = productsServiceProvider.get();
+
+      expect(result).toBe(instance(sparqueProductsServiceMock));
+    });
+
+    it('should return SparqueProductsService when Sparque config exists and no specific features are enabled (legacy mode)', () => {
+      configureTestBed({ apiKey: 'test-key' });
+      when(featureToggleServiceMock.getFeatures$('sparque_' as FeatureToggleType)).thenReturn(of([]));
+
+      const result = productsServiceProvider.get();
+
+      expect(result).toBe(instance(sparqueProductsServiceMock));
+    });
+
+    it('should return SparqueProductsService when Sparque config exists and multiple sparque features are enabled', () => {
+      configureTestBed({ apiKey: 'test-key' });
+      when(featureToggleServiceMock.getFeatures$('sparque_' as FeatureToggleType)).thenReturn(
+        of(['sparque_search', 'sparque_recommendations'] as FeatureToggleType[])
+      );
+
+      const result = productsServiceProvider.get();
+
+      expect(result).toBe(instance(sparqueProductsServiceMock));
+    });
+
+    it('should return ProductsService when Sparque config is null even if features are enabled', () => {
+      configureTestBed(undefined);
+      when(featureToggleServiceMock.getFeatures$('sparque_' as FeatureToggleType)).thenReturn(
+        of(['sparque_search'] as FeatureToggleType[])
+      );
+
+      const result = productsServiceProvider.get();
+
+      expect(result).toBe(instance(productsServiceMock));
+    });
+
+    it('should handle empty Sparque config object', () => {
+      configureTestBed({});
+      when(featureToggleServiceMock.getFeatures$('sparque_' as FeatureToggleType)).thenReturn(
+        of(['sparque_search'] as FeatureToggleType[])
+      );
+
+      const result = productsServiceProvider.get();
+
+      expect(result).toBe(instance(sparqueProductsServiceMock));
+    });
+
+    it('should use default parameter value for skipSparque when not provided', () => {
+      configureTestBed({ apiKey: 'test-key' });
+      when(featureToggleServiceMock.getFeatures$('sparque_' as FeatureToggleType)).thenReturn(
+        of(['sparque_search'] as FeatureToggleType[])
+      );
+
+      const result = productsServiceProvider.get();
+
+      expect(result).toBe(instance(sparqueProductsServiceMock));
+    });
+
+    it('should return ProductsService when skipSparque is explicitly false but conditions are not met', () => {
+      configureTestBed();
+
+      const result = productsServiceProvider.get(false);
+
+      expect(result).toBe(instance(productsServiceMock));
+    });
+  });
```

Comment:
test for isSparqueSearchEnabled ? - if this is implicitly done can you refactor it a bit and separate it?